### ISSUE TITLE
Cachi2 prefetching is making changes on a Go project's source

### ISSUE THREAD

**Issue by arewm** [hermetoproject/cachi2#707]
When running Cach2 on a source repo with the gomod package manager, changes are made to the `go.mod` file. This is presumably from running `go mod tidy`. It is also changing the sources when fetching the origin information for the git repository.

Cachi2 should not modify the sources that it is run on. Instead, it should operate on a temporary copy of the repository.

**Comment by eskultety** [comment]
Yes, this is known (or at least was common with projects requiring <=1.20 and having been processed with >=1.21). Recently we deprecated the `--gomod-vendor-check` and `--gomod-vendor` CLI flags that would have prevented us from fixing this (or at least not in a clean way). With those practically gone, we should be able to fix this in a simple and clean manner.
